pkg/storage/azure/azure_test.go

```diff
@@ -22,3 +37,137 @@ func TestGenerateAccountName(t *testing.T) {
 		}
 	}
 }
+
+func findEnvVar(envvars envvar.List, name string) *envvar.EnvVar {
+	for i, e := range envvars {
+		if e.Name == name {
+			return &envvars[i]
+		}
+	}
+	return nil
+}
+
+func TestConfigEnv(t *testing.T) {
+	ctx := context.Background()
+
+	cr := &imageregistryv1.Config{}
+	config := &imageregistryv1.ImageRegistryConfigStorageAzure{}
+
+	testBuilder := cirofake.NewFixturesBuilder()
+	testBuilder.AddInfraConfig(&configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: configv1.InfrastructureStatus{
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AzurePlatformType,
+				Azure: &configv1.AzurePlatformStatus{
+					ResourceGroupName: "resourcegroup",
+					CloudName:         configv1.AzureUSGovernmentCloud,
+				},
+			},
+		},
+	})
+	testBuilder.AddSecrets(&corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaults.CloudCredentialsName,
+			Namespace: defaults.ImageRegistryOperatorNamespace,
+		},
+		Data: map[string][]byte{
+			"azure_subscription_id": []byte("subscription_id"),
+			"azure_client_id":       []byte("client_id"),
+			"azure_client_secret":   []byte("client_secret"),
+			"azure_resourcegroup":   []byte("resourcegroup"),
+		},
+	})
+
+	listers := testBuilder.BuildListers()
+
+	authorizer := autorest.NullAuthorizer{}
+	sender := mocks.NewSender()
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"nameAvailable":true}`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`?`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"name":"account"}`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"keys":[{"value":"firstKey"}]}`))
+	sender.AppendResponse(mocks.NewResponseWithContent(`{"keys":[{"value":"firstKey"}]}`))
```

Comment:
Can't we use `AppendAndRepeatResponse` here?

Reply:
No, we should eliminate the second call and I'd like to have the second call recorded explicitly